src/components/Form/index.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
+
+import type {
+    Comment,
+    CommentReply,
+} from '../../state/comments';
+
+
+export interface CommentFormSetProps {
+    comments: Comment[];
+}
+
+export function CommentFormSetComponent(props: CommentFormSetProps) {
+    const [initialNumber] = React.useState(props.comments.filter(comment => {return comment.remoteId != null}).length);
```

Comment:
Yes, the reason why I left out the setter was that the important thing here was just storing the initial value. It's really just for efficiency, since it doesn't change. I can't find anything about a dynamic `initialState` causing it to be reset, but it looks like it might still recalculate the filtered array even if it doesn't use it - under 'lazy initial state' [here](https://reactjs.org/docs/hooks-reference.html) it suggests passing a function, and that it will only be evaluated once, so I'm not sure what the `useEffect` adds.

Haha, thanks, I do have a tendency to let my arrow functions contain more boilerplate than they actually need...

Re putting it in Redux store, I initially shied away from that because each `initialNumber` is only used in one component, but maybe that is the cleanest thing here. I'll have an experiment.

Reply:
Right, in the latest commits I've put it in the store instead. It's a bit of redundancy, but should speed things up a little.